code/modules/projectiles/guns/minigun.dm

```diff
@@ -0,0 +1,170 @@
+	if(ishuman(src.loc))
+		var/mob/living/carbon/human/H = src.loc
+		switch(over_object.name)
+			if("r_hand")
+				if(H.r_hand)
+					return
+				if(!H.unEquip(src))
+					return
+				H.put_in_r_hand(src)
+			if("l_hand")
+				if(H.l_hand)
+					return
+				if(!H.unEquip(src))
+					return
+				H.put_in_l_hand(src)
+	return
+
+/obj/item/weapon/minigunpack/update_icon()
+	if(armed)
+		icon_state = "minipack_n"
+	else
+		icon_state = "minipack_g"
+
+
+/obj/item/weapon/minigunpack/proc/attach_gun()
+	gun.forceMove(src)
+	armed = 0
+	update_icon()
+
+
+//The gun itself!
+/obj/item/weapon/gun/projectile/minigun
+	name = "M134 minigun"
+	desc = "She weighs one hundred fifty kilograms and fires two hundred dollar, custom-tooled cartridges at ten rounds per second."
+	icon = 'icons/obj/guns/minigun.dmi'
+	icon_state = "minigun_u"
+	item_state = "riotgun"
+	origin_tech = null
+	flags = NODROP
+	w_class = 5
+	materials = list()
+	burst_size = 10
+	pin = /obj/item/device/firing_pin/implant/pindicate
+	mag_type = /obj/item/ammo_box/magazine/internal/minigun
+	var/obj/item/weapon/minigunpack/ammo_pack
+	var/ready = 0
+
+/obj/item/weapon/gun/projectile/minigun/New()
+	if(!ammo_pack)
+		if(istype(loc,/obj/item/weapon/minigunpack)) //We should spawn inside a ammo pack so let's use that one.
+			ammo_pack = loc
+		else
+			qdel(src)//No pack, no gun
+	..()
+
+/obj/item/weapon/gun/projectile/minigun/can_shoot()
+	if(!ready)
+		return 0
+	else
+		return ..()
+
+/obj/item/weapon/gun/projectile/minigun/attack_self(mob/user)
+	if(ready)
+		unwield(user)
+	else
+		wield(user)
+
+/obj/item/weapon/gun/projectile/minigun/update_icon()
+	if(ready)
+		icon_state = "minigun_a"
+	else
+		icon_state = "minigun_u"
+
+/obj/item/weapon/gun/projectile/minigun/proc/unwield(mob/living/carbon/user)
+	if(!ready)
+		return
+	ready = 0
+	var/obj/item/weapon/twohanded/offhand/O = user.get_inactive_hand()
+	if(O && istype(O))
+		O.unwield()
+	update_icon()
+	return
+
+/obj/item/weapon/gun/projectile/minigun/proc/wield(mob/living/carbon/user)
+	ready = 1
+	if(user.get_inactive_hand())
+		user << "<span class='warning'>You need your other hand to be empty!</span>"
+		return
+	var/obj/item/weapon/twohanded/offhand/O = new(user) ////Let's reserve his other hand~
+	O.name = "[name] - offhand"
+	O.desc = "Your second grip on the [name]"
+	user.put_in_inactive_hand(O)
+	update_icon()
+	return
+
+/obj/item/weapon/gun/projectile/minigun/dropped(mob/user)
+	//handles unwielding a twohanded weapon when dropped as well as clearing up the offhand
+	if(user)
+		unwield(user)
+	ammo_pack.attach_gun()
+	return
+
+
+//All the weird BS that guns need to be as modular as possible. Oh yeah, the boolets.
+/obj/item/ammo_box/magazine/internal/minigun
+	name = "minigun belt"
+	ammo_type = /obj/item/ammo_casing/minigun
+	caliber = "7.62 NANO" //Technically the same ammo as the SAW but SHHHHH.
+	max_ammo = 3000
+
+/obj/item/ammo_casing/minigun
+	name = "a 7.62 NANO casing."
```

Comment:
I literally tell you which gun has it. But the minigun doesn't share for balance reasons.